clients/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogSegmentState.java

```diff
@@ -87,4 +89,27 @@ public byte id() {
     public static RemoteLogSegmentState forId(byte id) {
         return STATE_TYPES.get(id);
     }
+
+    public static boolean isValidTransition(RemoteLogSegmentState srcState, RemoteLogSegmentState targetState) {
+        Objects.requireNonNull(targetState, "targetState can not be null");
+
+        if (srcState == null) {
+            // If the source state is null, check the target state as the initial state viz DELETE_PARTITION_MARKED
+            // Wanted to keep this logic simple here by taking null for srcState, instead of creating one more state like
+            // COPY_SEGMENT_NOT_STARTED and have the null check by caller and pass that state.
+            return targetState == COPY_SEGMENT_STARTED;
+        } else if (srcState == targetState) {
```

Comment:
1 -> imho, this validation method should be part of the state enum and it can be used by any implementation including default RLMM. 

2 -> I would have preferred the suggested approach if there are many complex transitions but the transitions here are few and simple.